sunpy/visualization/animator/base.py

```diff
@@ -522,55 +500,112 @@ def _sanitize_axis_ranges(self, axis_ranges, data):
             * array of axis length, check that it was passed for a slider axes and do nothing
             if it was, error if it is not.
         """
+        ndim = len(data_shape)
         # If no axis range at all make it all [min,max] pairs
         if axis_ranges is None:
-            axis_ranges = [None] * data.ndim
+            axis_ranges = [None] * ndim
 
         # need the same number of axis ranges as axes
-        if len(axis_ranges) != data.ndim:
+        if len(axis_ranges) != ndim:
             raise ValueError("Length of axis_ranges must equal number of axes")
 
-        # For each axis validate and translate the axis_ranges
-        for i, d in enumerate(data.shape):
-            # If [min,max] pair or None
-            if axis_ranges[i] is None or (len(axis_ranges[i]) == 2 and d > 2):
-                # If min==max or None
-                if axis_ranges[i] is None or axis_ranges[i][0] == axis_ranges[i][1]:
-                    if i in self.slider_axes:
-                        axis_ranges[i] = np.arange(0, d+1)
-                    else:
-                        axis_ranges[i] = [0, d]
-                        # min max pair for slider axes should be converted
-                        # to an array
-                elif i in self.slider_axes:
-                    axis_ranges[i] = np.arange(axis_ranges[i][0], axis_ranges[i][1]+1)
-
-            # If we have a whole list of values for the axis, make sure we are a slider axis.
-            elif len(axis_ranges[i]) == d or axis_ranges[i].shape == data.shape:
-                axis_ranges[i] = np.array(axis_ranges[i])
-
-            # If above criteria are not met, raise an error.
+        # Define error message for incompatible axis_range input.
+        def incompatible_axis_ranges_error_message(j):
+            return ("Unrecognized format for {0}th entry in axis_ranges: {1}\n"
+                    "axis_ranges must be None, a [min, max] pair, or "
+                    "an array-like giving the edge values of each pixel, "
+                    "i.e. length must be length of axis + 1.".format(j, axis_ranges[j]))
+
+        # If axis range not given, define a function such that the range goes
+        # from -0.5 to number of pixels-0.5.  Thus, the center of the pixels
+        # along the axis will correspond to integer values.
+        def none_image_axis_range(j): return [-0.5, data_shape[j]-0.5]
+        # For each axis validate and translate the axis_ranges. For image axes,
```

Comment:
Can I get a link break here? It feels like that `none_image_axis_range` is just part of that large comment block.